src/DialogWindow.vala

```diff
@@ -496,10 +496,9 @@ namespace Ilia {
                 
                 if (entry.get_selection_bounds(out start, out end)) {
                     entry.get_buffer().delete_text(start, end - start);
-                    entry.get_buffer().insert_text(start, text.data);
+                    entry.insert_at_cursor(text);
```

Comment:
Given that this is repeated in both branches of the predicate, please move it out to the line below